Remove email waitlist

samsymons opened the pull request:
<!--
Note: This checklist is a reminder of our shared engineering expectations. Feel free to change it, although assigning a GitHub reviewer and the items in bold are required.
-->

Task/Issue URL: https://app.asana.com/0/0/1202144584700632/f
Tech Design URL:
CC:

**Description**:

This PR removes the email waitlist. See the Asana task for more.

<!--
Tagging instructions
If this PR isn't ready to be merged for whatever reason it should be marked with the `DO NOT MERGE` label (particularly if it's a draft)
If it's pending Product Review/PFR, please add the `Pending Product Review` label.

If at any point it isn't actively being worked on/ready for review/otherwise moving forward (besides the above PR/PFR exception) strongly consider closing it (or not opening it in the first place). If you decide not to close it, make sure it's labelled to make it clear the PRs state and comment with more information.
-->

**Steps to test this PR**:
- [ ] Install the app from develop
- [ ] Go to Settings -> Email and join the email waitlist
- [ ] Kill the app
- [ ] Install from this branch and launch
- [ ] App should not crash
- [ ] Go to Settings -> Email
- [ ] This should open a new tab taking you to `https://duckduckgo.com/email` (the webapp is not finished yet so it might redirect you to something different)
- [ ] Go to `https://duckduckgo.com/email/login` and sign in with your user
- [ ] In the browsing menu you should now see `Create Duck Address`
- [ ] Go to `github.com`, you should see the Dax icon in the email field
- [ ] Go to Settings and you should see your email address as the subtitle of the email row
- [ ] Click on the email settings and you should be taken to the email dashboard
- [ ] Click on Sign Out
- [ ] In the popup menu you should no longer see `Create Duck Address`
- [ ] Go to `github.com`, you should not see the Dax icon in the email field

<!--
Before submitting a PR, please ensure you have tested the combinations you expect the reviewer to test, then delete configurations you *know* do not need explicit testing.

Using a simulator where a physical device is unavailable is acceptable.
-->

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [X] iPhone 8
* [X] iPhone X
* [X] iPad

**OS Testing**:

* [ ] iOS 13
* [X] iOS 14
* [X] iOS 15

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**

samsymons committed:
Remove most of the email code.

samsymons committed:
Open the dashboard when tapping the settings row.

samsymons committed:
Remove email waitlist state on launch.

samsymons committed:
Remove unused app URLs.

samsymons committed:
Add a temporary check to tweak the user agent logic.

samsymons committed:
Remove the custom UA tweak.

samsymons committed:
Point to the quackdev email URL.

samsymons committed:
Remove unused email pixel definitions.

amddg44 reviewed (approved):
LGTM! 

Two small optional other waitlist related items that could be removed are these, but up to you 🙂 

1. https://github.com/duckduckgo/iOS/blob/b1ff95df06dd071136b03365e167422ef2d7c0c2/Core/UIColorExtension.swift#L120
2. A few image assets 
![image](https://user-images.githubusercontent.com/91189922/174124588-d4f24c2f-b92a-4fdc-af86-8341a14b40df.png)

samsymons committed:
Remove unused email waitlist assets.

samsymons committed:
Remove an email waitlist link color variable.

samsymons committed:
Merge branch 'develop' into sam/remove-email-waitlist

samsymons committed:
Merge branch 'develop' into sam/remove-email-waitlist

* develop:
  Fix Bookmark editing crash with Swipe-to-delete control (#1198)
  Update build number
  Update release notes
  Bump the version number
  Make secure vault errors more specific (#1212)
  Add Microsoft icon to network assets (#1211)
  Add Back/Forward navigation long press menu (#1199)
  Update internal user verification url to use-login.duckduckgo.com (#1210)
  Update SwiftLint workflow to always check all files in the repository (#1209)
  Autofill modal navigation changes (#1206)
  Bump build number
  Update App Store copy
  Silence linter warning
  Update embedded privacy config
  Bump version number

samsymons committed:
Fix a SwiftLint violation.

samsymons committed:
Merge branch 'develop' into sam/remove-email-waitlist

# By Fernando Bunn (13) and others
# Via Fernando Bunn (1) and amddg44 (1)
* develop: (23 commits)
  Updated embedded files
  Bump app version
  Check the image rule compilation error correctly (#1226)
  Fix iOS 15 dismiss ViewController problem. (#1178)
  Drop iOS 13 (#1208)
  Revert build version to 0
  Bump build version
  Ignore copyright date
  Bump build version
  Update metadata categories
  bump app version
  Update release version
  Revert "Move secure vault to group + pixels (#1217)" (#1224)
  Fix for crashes happening on killing the app (#1223)
  Fix typo (#1221)
  Update embedded files
  Update release notes
  Bump app version
  Block trackers on start (#1207)
  Update AMP deep extraction CBRs to block script load types (#1216)
  ...

# Conflicts:
#	Core/Pixel.swift
#	DuckDuckGo.xcodeproj/project.pbxproj
#	DuckDuckGo/EmailWaitlistWebViewController.swift

samsymons committed:
Resolve lingering merge issues.

samsymons committed:
Merge branch 'develop' into sam/remove-email-waitlist

# By Michal Smaga (5) and others
# Via Bartek Waresiak (6) and Michal Smaga (2)
* develop: (32 commits)
  Bump build number
  Update release notes
  Fix toggling protection from Menu (#1266)
  Update version number, release notes and metadata
  Infrastructure to expand 3rd-Party Tracker Loading Protection (#1265)
  Fix swiftlint warnings (#1264)
  Fix crash and item selection for Autofill Details items context menu (#1247)
  tie ActionMessageView to nav bar (#1214)
  Improving BarsAnimator component (#1237)
  Update embedded files
  Bump app version
  Email Manager keychain error reporting (#1258)
  Fix for decision handler in decidePolicyForNavigationResponse not being always called (#1259)
  Add assert and pixel for catching fail scenario when downloaded file is not found (#1257)
  fix downloads bug (#1256)
  Update BSK to latest (#1250)
  Fix menu button icon when repeat touches (#1254)
  prevent autofill suggestions when credentials have empty username (#1225)
  Update version and embedded files before release
  Resolve compilation condition if content blocking is globally disabled (#1253)
  ...

# Conflicts:
#	DuckDuckGo.xcodeproj/project.pbxproj
#	DuckDuckGo/Base.lproj/Settings.storyboard
#	DuckDuckGo/EmailProtectionViewController.swift
#	DuckDuckGo/EmailWaitlist.swift
#	DuckDuckGo/EmailWaitlistViewController.swift
#	DuckDuckGo/EmailWaitlistWebViewController.swift

samsymons committed:
Resolve post-merge issues.

samsymons committed:
Restore removed SwiftLint comments.

samsymons committed:
Organize the view controllers in the Settings storyboard.

With the email view controllers removed, we can bunch these up a little better.

samsymons committed:
Merge branch 'develop' into sam/remove-email-waitlist

samsymons committed:
Update the email URL.

samsymons committed:
Set BSK to 25.0.0.

samsymons committed:
Merge branch 'develop' into sam/remove-email-waitlist
